.github/workflows/CI.yaml

```diff
@@ -0,0 +1,258 @@
+name: Gitian CI
+
+on:
+  pull_request:
+    types:
+      - labeled
+
+jobs:
+  build-gitian:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: 'Set up Cloud SDK'
+      uses: 'google-github-actions/setup-gcloud@v2'
+      with:
+        version: '>= 363.0.0'
+
+    - name: Build Gitian
+      id: gitian
+      run: |
+        sudo apt update; sudo apt install wget openssh-client git -y
+        echo ${{ secrets.GCP_SA_KEY }} | base64 -d > json.json
+        gcloud auth activate-service-account --key-file=json.json
+
+        export random=$(cat /proc/sys/kernel/random/uuid | sed 's/[-]//g' | head -c 4; echo;)
```

Comment:
What happens if this clashes? If it just results in an error with probability $2^{-32}$ times the number of concurrent builds, that's fine I guess. If it could result in an incorrect build then I would prefer a different approach.

Reply:
In case the name matches, the machine cannot be generated, and the pipeline fails.

This is basically to ensure the unique identifier in GCP. But we can leave it at 12; I believe the number of characters for a VM name in GCP, if I remember correctly, is 32